### PR TITLE
Feature/connection bar

### PR DESCRIPTION
**Pull request by nbradbury**
Adds a "connection bar" to the main activity when a network connection isn't available. This relies on a new [ConnectionChangeReceiver](https://github.com/wordpress-mobile/WordPress-Android/blob/feature/connection_bar/WordPress/src/main/java/org/wordpress/android/networking/ConnectionChangeReceiver.java) which fires an event when the network availability changes. Any activity can use something like this to be alerted of the change:

```
@SuppressWarnings("unused")
public void onEventMainThread(ConnectionChangeReceiver.ConnectionChangeEvent event) {
	// event.isConnected() tells whether a connection is available
}
```

tngeoq1-nk

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/networking/ConnectionChangeReceiver.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import de.greenrobot.event.EventBus;
+
+/*
+ * global network connection change receiver - declared in the manifest to monitor
+ * android.net.conn.CONNECTIVITY_CHANGE
+ */
+public class ConnectionChangeReceiver extends BroadcastReceiver {
+
+    private boolean mWasConnected = true;
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/networking/ConnectionChangeReceiver.java]: Testing on HTC One with 5.1 I'm finding that a new ConnectionChangeReceiver is instantiated for each onReceive invocation which means the value of this is always true when onReceive is called. Can be resolved by making it static.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/networking/ConnectionChangeReceiver.java]: > Can be resolved by making it static.

Well that was silly of me. Fixed in 43ef614.

**Comment by tonyr59h** [comment]
Love this feature! Things like this are what make an app stand out :smiley:

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java [hunk 2]
```diff
@@ -445,6 +470,23 @@ public void onEventMainThread(NotificationEvents.NotificationsChanged event) {
...
+        updateConnectionBar(event.isConnected());
+    }
+
+    private void checkConnection() {
+        updateConnectionBar(NetworkUtils.isNetworkAvailable(this));
+    }
+
+    private void updateConnectionBar(boolean isConnected) {
+        if (isConnected && mConnectionBar.getVisibility() == View.VISIBLE) {
+            AniUtils.animateBottomBar(mConnectionBar, false);
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: When we're animating the connection bar can we apply the animation to the FAB as well? When testing the tap-to-retry functionality I noticed that it jumps as the connection bar slides in/out.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: > When we're animating the connection bar can we apply the animation to the FAB as well?

That would be a nice touch, but it's much trickier than it probably sounds. The FAB is on the "My Site" tab fragment and the connection bar is below the main activity's view pager which hosts the tab fragments. So as soon as the connection bar is made visible it "pushes" up the view pager, causing the jump you're seeing.

I could prevent this by having the connection bar overlay the view pager, but that means the bottom portion of each tab is obscured, which I think is a much worse problem.

Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: > I could prevent this by having the connection bar overlay the view pager, but that means the bottom portion of each tab is obscured, which I think is a much worse problem.

That's actually the first thing I verified during the review, would not be good to obstruct the view. Good reason for not implementing the animation :+1:

:shipit:

**Pull request merged**